src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -84,53 +85,57 @@ private void saveModificationsNonTransactional(UUID groupUuid, List<? extends Mo
+        // load origin entities (metadata light mode)
+        List<ModificationEntity> modificationEntities = getModificationsMetadataEntities(originGroupUuid, false);
+        Map<UUID, ModificationEntity> originModifications = modificationEntities.stream()
                 .collect(Collectors.toMap(ModificationEntity::getId, Function.identity(), (x, y) -> y, LinkedHashMap::new));
-
         List<UUID> modificationsToMoveUUID = modificationsUuid.stream().filter(originModifications::containsKey).collect(Collectors.toList());
 
-        List<ModificationEntity> newDestinationModificationList = List.of();
-        if (!modificationsToMoveUUID.isEmpty()) {
-            // if moving within the same group
-            if (originGroupUuid.equals(destinationGroupUuid)) {
-                if (referenceModificationUuid != null && !originModifications.containsKey(referenceModificationUuid)) {
-                    throw new NetworkModificationException(MOVE_MODIFICATION_ERROR);
-                }
-
-                newDestinationModificationList = updateModificationList(modificationsToMoveUUID, originModifications, originModifications, referenceModificationUuid);
-
-                originModificationGroupEntity.setModifications(newDestinationModificationList);
-            } else {
-                //if destination is empty, group does not exist, we create it here if needed
-                ModificationGroupEntity destinationModificationGroupEntity = getOrCreateModificationGroup(destinationGroupUuid);
-
-                Map<UUID, ModificationEntity> destinationModifications = getModificationsEntities(destinationGroupUuid).stream()
-                        .collect(Collectors.toMap(ModificationEntity::getId, Function.identity(), (x, y) -> y, LinkedHashMap::new));
-
-                // referenceModificationUuid must belong to destination one
-                if (referenceModificationUuid != null && !destinationModifications.containsKey(referenceModificationUuid)) {
-                    throw new NetworkModificationException(MOVE_MODIFICATION_ERROR);
-                }
-
-                newDestinationModificationList = updateModificationList(modificationsToMoveUUID, originModifications, destinationModifications, referenceModificationUuid);
-
-                originModificationGroupEntity.setModifications(new ArrayList<>(originModifications.values()));
-                destinationModificationGroupEntity.setModifications(newDestinationModificationList);
+        if (originGroupUuid.equals(destinationGroupUuid)) { // single group case
+            if (referenceModificationUuid != null && !originModifications.containsKey(referenceModificationUuid)) {
+                throw new NetworkModificationException(MOVE_MODIFICATION_ERROR);
+            }
+            // re-organize the list
+            updateModificationsOrder(updateModificationList(modificationsToMoveUUID, originModifications, referenceModificationUuid));
```

Comment:
Here it's a bit weird that updateModificationList() modify the originModifications map AND at the same time, updateModificationList() is mostly used for its returned value.

I feel like this method could return the updated list without modiying the parameters.
And updateModificationList() and insertModifications() could work with lists as parameters and not with a map.

Reply:
yes, I tried to change/rename things a bit for clarity.